pkg/controller/operators/catalog/operator.go

```diff
@@ -1062,6 +1142,15 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 		logger.Debugf("no subscriptions were updated")
 	}
 
+	// Make sure that we no longer indicate unpacking progress
```

Comment:
Apologies if I'm not understanding the order of operations correctly here, but does this condition not become true _before_ calling `ensureInstallPlan`? Or in other words shouldn't these status updates come on line 1110?  

Or else, are we saying that a bundle unpack isn't successful until an `InstallPlan` is successfully created with the unpacked bundles?

Reply:
> Apologies if I'm not understanding the order of operations correctly here, but does this condition not become true _before_ calling `ensureInstallPlan`? Or in other words shouldn't these status updates come on line 1110?

Your understanding is correct. Order is as follows:
* We first create an unpack job within `o.unpackBundles(...)`
* Based on the result from the above we either set `SubscriptionBundleUnpackFailed` (in case of errors) or `SubscriptionBundleUnpacking` (in case when we still waiting on the unpack job) to `True`. I believe `SubscriptionBundleUnpacking` will be the most common on the first run when syncing with a new bundle.
* Once unpacking is done - we create an `InstallPlan` in `o.ensureInstallPlan(...)`
  * We just create an `InstallPlan` here, but syncing of the `InstallPlan` is a separate process - (see `o.syncInstallPlans`)
* All being well - we make sure to remove `SubscriptionBundleUnpackFailed` on subs (if any exist) and we set `SubscriptionBundleUnpacking` to `False`.
* Update subs


> Or else, are we saying that a bundle unpack isn't successful until an `InstallPlan` is successfully created with the unpacked bundles?

Technically unpack can be successfull without successfull `InstallPlan` creation. But in terms of current implementation in this PR - you are right.

I considered removing `SubscriptionBundleUnpackFailed` and setting `SubscriptionBundleUnpacking` to `False` before `o.ensureInstallPlan(...)`, but it makes syncing a bit more complex becaues we would have to do two updates to the subs. We would have to implement something like this:

* Unpacking (first two steps from the list above)
* Removing `SubscriptionBundleUnpackFailed` and setting `SubscriptionBundleUnpacking` to `False`
* `o.updateSubscriptionStatuses(subs)` <--- First update
* `o.ensureInstallPlan(...)` and `o.setIPReference(...)`
* `o.updateSubscriptionStatuses(subs)` <--- Second update

Instead we just do:

* Unpacking
* `o.ensureInstallPlan(...)` and `o.setIPReference(...)`
* Removing `SubscriptionBundleUnpackFailed` and setting `SubscriptionBundleUnpacking` to `False`
* `o.updateSubscriptionStatuses(subs)` <--- The only update to subs at the end

So it is a bit of optimisation, but I'm happy to reconsider. Let me know if you see an issue with this.

Reply:
Thanks for answering, I appreciate the level of detail!  
I'm fine with what you have here to keep API calls to a minimum. We _probably_ don't need such a fine-grained level of tracing for the process anyway - I mostly wanted to make sure there was intention behind it. We can keep it the way you have it :)